manifests/controller/config.pp

```diff
@@ -89,134 +64,142 @@
-  Enum['debug', 'info', 'notice', 'warn', 'error']     $log_file_level     = 'info',
-  Optional[Boolean]                                    $log_file_append    = undef,
-  Boolean                                              $disable_analytics  = true,
+  # Local Target Directory Options
+  Optional[String[1]]              $local_group        = $simp_bolt::controller::local_group_name,
+  Stdlib::Unixpath                 $local_home         = pick($simp_bolt::controller::local_user_home, '/var/local/simp_bolt'),
+  Optional[String[1]]              $local_user         = $simp_bolt::controller::local_user_name,
+
+  # Config File Specification
+  Optional[Hash]                   $config_hash        = undef,
+
+  # Global Bolt Options
+  Boolean                          $color              = true,
+  Optional[Integer[0]]             $concurrency        = undef,
+  Simp_bolt::Transport             $default_transport  = 'ssh',
+  Boolean                          $disable_analytics  = true,
+  Optional[Enum['human','json']]   $format             = undef,
+  Optional[String[1]]              $hiera_config       = undef,
+  Optional[String[1]]              $inventoryfile      = undef,
+  Simp_bolt::LogLevel              $log_console_level  = 'info',
+  Boolean                          $log_file_append    = false,
+  Simp_bolt::LogLevel              $log_file_level     = 'info',
+  Stdlib::Unixpath                 $log_file           = '/var/log/puppetlabs/bolt/bolt.log',
+  Optional[String[1]]              $modulepath         = undef,
+
+  # Overall Transport Options
+  Hash[Simp_bolt::Transport, Hash] $transport_options  = {
+                                     'ssh' => {
+                                       'tmpdir' => '/var/local/bolt_user',
+                                       'user'   => 'bolt_user',
+                                       'run-as' => 'root'
+                                     }
+                                   }
 ){
   assert_private()
 
-  # If local user and home directory are specified create the Boltdir and its parent directory
-  if $local_user and $local_home {
-    exec { "mkdir -p ${local_home}":
-      path   => ['/bin','/usr/bin'],
-      onlyif => "test ! -d ${local_home}"
-    }
-
-    $_bolt_dir_parent = "${local_home}/.puppetlabs"
-    $_bolt_dir = "${local_home}/.puppetlabs/bolt"
-
-    file { $_bolt_dir_parent:
-      ensure => 'directory',
-      owner  => $local_user,
-      group  => $local_group,
-      mode   => '0750'
-    }
-
-    file { $_bolt_dir:
-      ensure => 'directory',
-      owner  => $local_user,
-      group  => $local_group,
-      mode   => '0750'
-    }
-
-    # Create the config file for bolt
-    file { "${_bolt_dir}/bolt.yaml":
-      ensure  => 'file',
-      owner   => $local_user,
-      group   => $local_group,
-      mode    => '0640',
-      content => template("${module_name}/bolt_yaml.erb")
-    }
-  } else {
+  unless $transport_options[$default_transport] {
```

Comment:
This should include an or in case the entire file is specified via $config_hash.